app/src/main/java/com/randomappsinc/studentpicker/database/DataSource.java

```diff
@@ -134,6 +134,38 @@ public void renameList(int listId, String newName) {
         close();
     }
 
+    public Map<ListDO, List<NameDO>> getAllListSet() {
+        Map<ListDO, List<NameDO>> allListMap = new HashMap<>();
+        ListDO listDO = new ListDO(-1, "");
+        List<NameDO> nameDOList = new ArrayList<>();
+
+        open();
+        String query = "SELECT Lists.id, Lists.list_name, Names.name, Names.name_count, Names.photo_uri " +
+                "FROM Lists " +
+                "LEFT JOIN Names ON Lists.id = Names.list_id " +
+                "ORDER BY Lists.id";
+        Cursor cursor = database.rawQuery(query, null);
+        while (cursor.moveToNext()) {
+            int listId = cursor.getInt(0);
+            String listName = cursor.getString(1);
+            if (listDO.getId() != listId){
+                listDO = new ListDO(listId, listName);
+                nameDOList = new ArrayList<>();
+            }
+            NameDO nameDO = new NameDO();
+            String name = cursor.getString(2);
+            nameDO.setName(name);
+            int nameAmount = cursor.getInt(3);
+            nameDO.setAmount(nameAmount);
+            nameDO.setPhotoUri(cursor.getString(4));
```

Comment:
I'm not sure if the photo URI will ever work when restoring from backup since it's app/device specific I think. If it doesn't work, we should just not back this up, similar to flashcards.